### ISSUE TITLE
Allow xxxx in date field when adding new edition

### ISSUE THREAD

**Issue by seabelis** [internetarchive/openlibrary#5254]
If an edition has no date, Xs may be used as placeholders according to our documentation regarding publication dates. However, when adding a new edition, only numbers are allowed.

### Evidence / Screenshot (if possible)
<img width="864" alt="Screenshot 2021-06-01 at 19 40 04" src="https://user-images.githubusercontent.com/17739465/120367454-2ce50900-c311-11eb-9e19-672e7f8b5feb.png">
### Steps to Reproduce

1. Go to ...https://openlibrary.org/books/add
2. Do ...use Xs in date field

* Actual: Error
* Expected: Xs allowed

### Details

- **Logged in (Y/N)?** Y
- **Browser type/version?** Version 91.0.4472.77 (Official Build) (x86_64)
- **Operating system?** MacOS Big Sur
- **Environment (prod/dev/local)?** prod

### Stakeholders

@cdrini

**Comment by RayBB** [comment]
@seabelis would it be preferable to allow it to be blank on the website and then on the server side add the xxxx automatically?

**Comment by seabelis** [comment]
I don't think so. That would populate all blank years with xxxx rather than those specific items without a date. Blank just means it hasn't been filled in, not that the item is undated.

**Comment by RayBB** [comment]
In that case, what would you think about adding a checkbox for "undated"?
That may be a bit easier for folks without having to read docs to learn about the xxxx.

**Comment by seabelis** [comment]
Such a checkbox would be fine, but allowing Xs in the field should also be possible for those who have read the documentation. Or for situations where date is approximate, 19xx, for example. Ultimately, the formats used on the edition form should not be different from the "add a book" form.

**Comment by ClementineAccount** [comment]
@scottbarnes I could attempt this within the week once assigned.

**Comment by ClementineAccount** [comment]
I was doing some testing on local to get a better understanding of the issue and imported a test book using command line. 
I need some confirmation and clarification on what the 'xxxx' publish date should be but I think this seems correct. If so I can now add the additional check in the validator under validate.js to add support for 'xxxx' entries in the regex.
![image](https://github.com/internetarchive/openlibrary/assets/26779639/1c320162-7d6b-4d98-a086-4d32ab419af7)

(So then I can add an additional check below in validate.js to support 'xxxx' regex)
```
    // validate publish-date to make sure the date is not in future
    // used in templates/books/add.html
    jQuery.validator.addMethod('publish-date', function(value) {
        // if it doesn't have even three digits then it can't be a future date
        var tokens = /(\d{3,})/.exec(value);

        var year = new Date().getFullYear();
        return tokens && tokens[1] && parseInt(tokens[1]) <= year + 1; // allow one year in future.
    },
    'Are you sure that\'s the published date?'
    );
```

**Comment by scottbarnes** [comment]
@ClementineAccount, yes, that publication date of `xxxx` from the screenshot is what it should look like, `validate.js` looks like the correct place to update the regex. :)

**Comment by ClementineAccount** [comment]
I created the [Draft pull request](https://github.com/internetarchive/openlibrary/pull/8664#issue-2054685125). 

Currently I need some clarification on the acceptable 4 digit dates because I am unsure if 'xx99' or 'x999' dates look correct. I think this means that the first character must always be a digit, unless it is 'xxxx'.

**Comment by agmckee** [comment]
I just ran into this issue. I wanted to add a book that I know about from a reference, but for which there is no publication date or publisher. It is impossible to add such a book without entering data for the publisher, which is unknown, or the publication date. This seems contrary to Open Library's stated goals of having a web page for every book; not all books have been published or have data available for the publication date.

**Comment by scottbarnes** [comment]
@agmckee, there's a draft PR to address this that we can hopefully get over the line soon.

**Comment by agmckee** [comment]
This approach seems misguided to me. I am not a librarian but from a data perspective, it seems wrong to have incomplete dates. I realize that the documentation apparently suggests this, but to me it seems wrong.

What is the end-user supposed to infer if a date is shown as `19xx`? That the book was published in the 20th century? Or that the librarian/record-enterer _believed_ the book was published in the 20th century? 

Wouldn't it be preferable just to allow adding books without a publication date or a publisher? Such books exist in libraries throughout the world, books that were started but not completed but made available in manuscript form in libraries and archives, books that were published but without a publication date being put in the book (lots of early American works don't have a publication date), books published in antiquity sans date or publisher information.

Putting data in a system that is wrong or not supported by evidence just seems really wrong-headed to me. If data is unknown, it should be `null` values to reflect that it is not known.

**Comment by scottbarnes** [comment]
With respect to unknown publication dates, the plan is to allow for one to enter books without a publication date by entering `xxxx`, following the format outlined here: https://openlibrary.org/help/faq/editing#publication-date.

However, I understand that doesn't address the concern that format Open Library follows, [EDTF](https://www.loc.gov/standards/datetime/) allows for publication dates circa some date, such as `1953?`, or with one date missing, such as `195X`.

Whether Open Library should follow another format that isn't EDTF is a fair question I think, but with respect to this issue, the goal is to bring the date entry form into alignment with the currently-followed date format. But that doesn't mean the debate shouldn't be had, albeit in another issue.

**Comment by seabelis** [comment]
xxxx means the item has no date. Libraries sometimes indicate this as n.d. or uuuu. They mean the same thing. We use xxxx. A null value does not indicate the item has no date; it indicates we have no data.
